sklearn/multioutput.py

```diff
@@ -359,3 +362,228 @@ def score(self, X, y):
+        the label matrix Y.
+            order = [0, 1, 2, ..., Y.shape[1] - 1]
+
+        The order of the chain can be explicitly set by providing a list of
+        integers. For example, for a chain of length 5
+            order = [1, 3, 2, 4, 0]
+        means that the first model in the chain will make predictions for
+        column 1 in the Y matrix, the second model will make predictions
+        for column 3, etc.
+
+        If order is 'random' a random ordering will be used.
+
+    cv : int, cross-validation generator or an iterable, optional
+        Determines whether to use cross validated predictions or true
+        labels for the results of previous estimators in the chain.
+        If cv is None the true labels are used when fitting. Otherwise
+        possible inputs for cv are:
+            integer, to specify the number of folds in a (Stratified)KFold,
+            An object to be used as a cross-validation generator.
+            An iterable yielding train, test splits.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`.
+
+        Used to generate random chain orders.
+
+    Attributes
+    ----------
+    classes_ : list of arrays of length len(estimators_) containing the
+        class labels for each estimator in the chain.
+
+    estimators_ : list
+        A list of copies of base_estimator. Once fit the estimators in
+        this list will be ordered as specified by the order attribute.
+        The original order of labels (as specified by Y) is recovered in
+        the predict method by indexing into the predictions with the
+        list of indices in the order attribute.
+
+
+    References
+    ----------
+    Jesse Read, Bernhard Pfahringer, Geoff Holmes, Eibe Frank, "Classifier
+    Chains for Multi-label Classification", 2009.
+
+    """
+
+    def __init__(self, base_estimator, order=None, cv=None, random_state=None):
+        self.base_estimator = base_estimator
+        self.order = order
+        self.cv = cv
+        self.random_state = random_state
+
+    def fit(self, X, Y):
+        """Fit the model to data matrix X and targets Y.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+        Y : {array-like, sparse matrix}, shape (n_samples, n_classes)
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+
+        X, Y = check_X_y(X, Y,
+                         multi_output=True,
+                         accept_sparse=True)
+
+        random_state = check_random_state(self.random_state)
+
+        if self.order is None:
+            self.order = np.array(range(Y.shape[1]))
```

Comment:
we do not overwrite parameters, by convention. Instead, store an attribute `order_`.